refactor(parser): add debug checks that `Modifiers` is consistent

overlookmotel opened the pull request:
Add debug assertions to ensure that the 2 representations of modifiers in `Modifiers` are in sync with each other. It'd be easy to make a mistake in the calling code.

Boshen commented:
### Merge activity

* **Nov 17, 3:16 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 17, 3:16 PM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
refactor(parser): add debug checks that `Modifiers` is consistent (#15784)

Add debug assertions to ensure that the 2 representations of modifiers in `Modifiers` are in sync with each other. It'd be easy to make a mistake in the calling code.
